VMware documentation changes (#1984)
hashicorp/terraform-website: content/source/docs/enterprise/before-installing/reference-architecture/vmware.md

### Failure Scenarios

VMware vSphere provides a high level of resilience in various cases
of failure (at the server hardware layer through vSphere High Availablilty (HA) and at the network layer through virtual distributed
switching.) In addition, having workloads failover to a DR datacenter with tools like VMware Site Recovery Manager or utilizing stretched clusters

> provides recovery in the case of a total data center outage. See the Disaster Recovery section.

Author: 1. HA is the backing technology that is set on a vSphere cluster that restarts VMs on a designated physical ESXi host if the one that they are running on fails. vMotion only happens if DRS (Distributed Resources Scheduler) has a recommendation to migrate the workload to a host that has more available resources, a user manually migrates the workload, or the host that the workload was on was put in maintenance mode. 

2. I would say that HA is a requirement and this is enabled in most if not all vSphere clusters. 

3. I always refer to VMs as workloads but we can rephrase this to something else more specific if you would like. 

4. The callout to SRM or other products is really to have the customer be aware that they can use 3rd party tools to replicate the LUN where the VMs reside or the VMs themselves so that way they can have automated failover processes which replicate the running state of the workloads. This handles all of the replication.

> VMware vSphere provides a high level of resilience in various cases
> of failure (at the server hardware layer through vSphere High Availablilty (HA) and at the network layer through virtual distributed
> switching.) In addition, having workloads failover to a DR datacenter with tools like VMware Site Recovery Manager or utilizing stretched clusters
> provides recovery in the case of a total data center outage. See the Disaster Recovery section.

Reviewer: ```suggestion
VMware vSphere provides a high level of resilience in various cases
of failure, such as at the server hardware layer through vSphere High Availability (HA) and at the network layer through virtual distributed
switching. In addition, employing tools such as VMware Site Recovery Manager or utilizing stretched clusters
can assist in recovery in the case of a total data center to support failover to a DR datacenter. See the Disaster Recovery section.
```

> provides recovery in the case of a total data center outage. See the Disaster Recovery section.

Reviewer: Did a little rewording to soften the impression that an automatic DR failover can always be done just through the facilities and to better highlight add-on tools would be needed.

> Additional failover can be provide by the Active/Active installation mode.

Reviewer: ```suggestion
Additional failover can be provided by the Active/Active installation mode.
```

### Failure Scenarios

VMware vSphere provides a high level of resilience in various cases
of failure (at the server hardware layer through vSphere High Availablilty (HA) and at the network layer through virtual distributed
switching.) In addition, having workloads failover to a DR datacenter with tools like VMware Site Recovery Manager or utilizing stretched clusters
provides recovery in the case of a total data center outage. See the Disaster Recovery section.

> Additional failover can be provided by the Active/Active installation mode.

Reviewer 2: ```suggestion
The Active/Active installation mode can provide additional failover.
```

be recovered, or traffic can be routed to the *TFE-standby* server to
resume service when the failure is limited to the Terraform Enterprise server layer. See the Disaster Recovery section.

#### Single ESXi Host Failure

In the event of a single ESXi host failure, vSphere HA will restart the Terraform Enterprise virtual
machine to a functioning ESXi host in the cluster. This restart can take up to 30 seconds for the failed virtual machine to come back online on a healthy host within the cluster.

> If VMware vSphere Fault Tolerance (FT) has been configured for the TFE server, the failover does not result in any visiable outage to the end user.

Author: Yes this would be based on the general recommendation that the customer use vSphere HA at the cluster (This is a VMware base recommendation for any compute cluster) and automated restarts would take place if a host fails. 

We mentioned FT in the original doc (I have rarely seen people use it when I worked at VMware) so I kept it as a 0 downtime type option but this would really be a corner case.

> If VMware vSphere Fault Tolerance (FT) has been configured for the TFE server, the failover does not result in any visiable outage to the end user.

Reviewer: OK, that answered my concerns.
